network/proc/findpid.go

```diff
@@ -63,9 +57,11 @@ func findPID(uid, inode int) (pid int) {
 
 		// If we have found PIDs, search them.
 		if ok {
-			for _, pid = range pids {
-				if findSocketFromPid(pid, socketName) {
-					return pid
+			// Look through the PIDs in reverse order, because higher/newer PIDs will be more likely to
+			// be searched for.
+			for i := len(pids) - 1; i > 0; i-- {
+				if findSocketFromPid(pids[i], socketName) {
+					return pids[i]
```

Comment:
```suggestion
			for i := len(pids) - 1; i >= 0; i-- {
				if findSocketFromPid(pids[i], socketName) {
					return pids[i]
```
Same issue a few lines below